### PR TITLE
fix: don't have required columns if they have computed

### PR DESCRIPTION
**Pull request by chinook25**
how to test:
- explain here what to do to test this (or point to unit tests)

todo:
- [ ] updated docs in case of new feature
- [ ] added tests

### CODE REVIEW

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java [hunk 1]
```diff
@@ -249,21 +249,33 @@ public Column setReadonly(Boolean readonly) {
...
+      this.required = required.toString();
+    } else {
+      this.required = "false";
+    }
     return this;
   }
 
   public Column setRequired(String required) {
-    this.required = required;
+    if (this.computed == null) {
```
Comment (mswertz) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java:265]: I think we don't need this, the 'get' will solve all

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java [hunk 2]
```diff
@@ -249,21 +249,33 @@ public Column setReadonly(Boolean readonly) {
   }
 
   public boolean isRequired() {
-    return required != null && required.equalsIgnoreCase("true");
+    return this.computed == null && required != null && required.equalsIgnoreCase("true");
   }
 
   public Column setRequired(Boolean required) {
-    this.required = required.toString();
+    if (this.computed == null) {
```
Comment (mswertz) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/Column.java:256]: I think we don't need this, the 'get' will solve all

**Review by mswertz — COMMENTED** [review]
The failing test shows an issue in our thinking: primary key fields are always set to be required. How can we have a computed field that is also a primary key field?

The AUTO_ID example is the issue in pet store, which is also a primary key but in our wisdom we also made it required.
See https://github.com/molgenis/molgenis-emx2/blob/86ccb875f18ead7648db9a0d2f9a00fc0810cb28/backend/molgenis-emx2-datamodels/src/main/java/org/molgenis/emx2/datamodels/PetStoreLoader.java#L52

Which makes me wonder if we are on the right track. I mean, there should not be an issue to make a 'computed' field also 'required'. We only should make sure that the 'required' is checked after the 'computed'.

**Comment by chinook25** [comment]
Needs more thinking before doing this.

**Pull request closed without merging**